cmd/argoexec/commands/emissary.go

```diff
@@ -0,0 +1,216 @@
+				}
+			}()
+
+			// this also indicates we've started
+			if err := os.MkdirAll(varArgo+"/ctr/"+containerName, 0700); err != nil {
+				return fmt.Errorf("failed to create ctr directory: %w", err)
+			}
+
+			name, args := args[0], args[1:]
+
+			signals := make(chan os.Signal, 1)
+			defer close(signals)
+			signal.Notify(signals)
+			defer signal.Reset()
+			go func() {
+				for s := range signals {
+					if s != syscall.SIGCHLD {
+						_ = syscall.Kill(-os.Getpid(), s.(syscall.Signal))
+					}
+				}
+			}()
+
+			data, err := ioutil.ReadFile(varArgo + "/template")
+			if err != nil {
+				return fmt.Errorf("failed to read template: %w", err)
+			}
+
+			if err := json.Unmarshal(data, template); err != nil {
+				return fmt.Errorf("failed to unmarshal template: %w", err)
+			}
+
+			name, err = path.Search(name)
+			if err != nil {
+				return fmt.Errorf("failed to find name in PATH: %w", err)
+			}
+
+			command := exec.Command(name, args...)
+			command.Env = os.Environ()
+			command.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+			stdout, err := os.Create(varArgo + "/ctr/" + containerName + "/stdout")
+			if err != nil {
+				return fmt.Errorf("failed to open stdout: %w", err)
+			}
+			defer func() { _ = stdout.Close() }()
+			command.Stdout = io.MultiWriter(os.Stdout, stdout)
+
+			stderr, err := os.Create(varArgo + "/ctr/" + containerName + "/stderr")
+			if err != nil {
+				return fmt.Errorf("failed to open stderr: %w", err)
+			}
+			defer func() { _ = stderr.Close() }()
+			command.Stderr = io.MultiWriter(os.Stderr, stderr)
+
+			if err := command.Start(); err != nil {
+				return err
+			}
+
+			go func() {
+				for {
+					data, _ := ioutil.ReadFile(varArgo + "/ctr/" + containerName + "/signal")
+					_ = os.Remove(varArgo + "/ctr/" + containerName + "/signal")
+					s, _ := strconv.Atoi(string(data))
+					if s > 0 {
+						_ = syscall.Kill(command.Process.Pid, syscall.Signal(s))
+					}
+					time.Sleep(2 * time.Second)
+				}
+			}()
+
+			cmdErr := command.Wait()
+
+			if cmdErr == nil {
+				exitCode = 0
+			} else if exitError, ok := cmdErr.(*exec.ExitError); ok {
+				if exitError.ExitCode() >= 0 {
+					exitCode = exitError.ExitCode()
+				} else {
+					exitCode = 137 // SIGTERM
+				}
+			}
+
+			if err := stderr.Close(); err != nil {
+				return fmt.Errorf("failed to close stderr: %w", err)
+			}
+			if err := stdout.Close(); err != nil {
+				return fmt.Errorf("failed to close stdout: %w", err)
+			}
+
+			if containerName == common.MainContainerName {
+				for _, x := range template.Outputs.Parameters {
+					if x.ValueFrom != nil && x.ValueFrom.Path != "" {
+						if err := saveParameter(x.ValueFrom.Path); err != nil {
+							return err
+						}
+					}
+				}
+				for _, x := range template.Outputs.Artifacts {
+					if x.Path != "" {
+						if err := saveArtifact(x.Path); err != nil {
+							return err
+						}
+					}
+				}
+			} else {
+				println("not saving outputs - not main container")
```

Comment:
this will be mingled with the containers logs, it might be good to make it conventional, e.g.

```
#argo msg=not saving outputs
```